pyproject.toml

```diff
@@ -1,5 +1,12 @@
 [build-system]
-requires = ["setuptools>=67.8", "wheel", "cython~=3.0.2", "oldest-supported-numpy"]
+requires = [
+    "setuptools>=67.8",
+    "wheel",
+    "cython~=3.0.2",
+    "numpy>=2.0.0rc1; python_version >= '3.9'",
```

Comment:
I have a different reason for suggesting this change. If numpy has future pre-releases, it could cause instability when building rasterio from source with this pin as it would use that version in the build.

```suggestion
    "numpy==2.0.0rc1; python_version >= '3.9'",
```

Reply:
Will do!